edit/networking.k8s.io.networkpolicy/PolicyRuleTarget.vue

```diff
@@ -0,0 +1,281 @@
+      this.$nextTick(() => {
+        this.$set(this.value, TARGET_OPTION_IP_BLOCK, {});
+      });
+    }
+
+    const matchingPods = this.getMatchingPods();
+    const matchingNamespaces = this.getMatchingNamespaces();
+
+    return {
+      portOptions:   ['TCP', 'UDP'],
+      matchingPods,
+      matchingNamespaces,
+      invalidCidrs:    [],
+      TARGET_OPTION_IP_BLOCK,
+      TARGET_OPTION_NAMESPACE_SELECTOR,
+      TARGET_OPTION_POD_SELECTOR,
+      POD,
+      targetOptions: [
+        TARGET_OPTION_IP_BLOCK,
+        TARGET_OPTION_NAMESPACE_SELECTOR,
+        TARGET_OPTION_POD_SELECTOR
+      ],
+    };
+  },
+  computed: {
+    podSelectorExpressions: {
+      get() {
+        return convert(
+          this.value[TARGET_OPTION_POD_SELECTOR]?.matchLabels || {},
+          this.value[TARGET_OPTION_POD_SELECTOR]?.matchExpressions || []
+        );
+      },
+      set(podSelectorExpressions) {
+        this.$set(this.value, TARGET_OPTION_POD_SELECTOR, simplify(podSelectorExpressions));
+      }
+    },
+    namespaceSelectorExpressions: {
+      get() {
+        return convert(
+          this.value[TARGET_OPTION_NAMESPACE_SELECTOR]?.matchLabels || {},
+          this.value[TARGET_OPTION_NAMESPACE_SELECTOR]?.matchExpressions || []
+        );
+      },
+      set(namespaceSelectorExpressions) {
+        this.$set(this.value, TARGET_OPTION_NAMESPACE_SELECTOR, simplify(namespaceSelectorExpressions));
+      }
+    },
+    selectTargetOptions() {
+      const selectTargetOptions = [];
+
+      for (const option of this.targetOptions) {
+        selectTargetOptions.push({
+          label: this.t(`networkpolicy.rules.${ option }.label`),
+          value: option,
+        });
+      }
+
+      return selectTargetOptions;
+    },
+    targetType: {
+      get() {
+        for (const option of this.targetOptions) {
+          if (this.value[option]) {
+            return option;
+          }
+        }
+
+        return null;
+      },
+      set(targetType) {
+        this.$delete(this.value, TARGET_OPTION_IP_BLOCK);
+        this.$delete(this.value, TARGET_OPTION_NAMESPACE_SELECTOR);
+        this.$delete(this.value, TARGET_OPTION_POD_SELECTOR);
+        this.$nextTick(() => {
+          this.$set(this.value, targetType, {});
+        });
+      }
+    }
+  },
+  watch: {
+    namespace:                    'updateMatches',
+    'value.podSelector':          'updateMatches',
+    'value.namespaceSelector':    'updateMatches',
+    'value.ipBlock.cidr':         'validateCIDR',
+    'value.ipBlock.except':       'validateCIDR',
+    podSelectorExpressions:       'updateMatches',
+    namespaceSelectorExpressions: 'updateMatches',
+  },
+  methods: {
+    validateCIDR() {
+      const exceptCidrs = this.value[TARGET_OPTION_IP_BLOCK].except || [];
+
+      this.invalidCidrs = [];
+
+      const regex = /^((25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\/(3[0-2]|2[0-9]|1[0-9]|[0-9])$/;
```

Comment:
Although overkill for this use case, validating a cidr might be useful elsewhere (i think the harvester team have hit this). Could this be made into a `@/utils/validators/cidr.js` so it's in a common place ready to use next time?

Reply:
Done